core/types.ml

```diff
@@ -133,11 +158,20 @@ type typ =
     | `Variant of row
     | `Effect of row
     | `Table of typ * typ * typ
+    | `Lens of lens_sort 
     | `Alias of ((string * type_arg list) * typ)
     | `Application of (Abstype.t * type_arg list)
     | `MetaTypeVar of meta_type_var
     | `ForAll of (quantifier list ref * typ)
     | (typ, row) session_type_basis ]
+and lens_sort     = fundepset * lens_phrase option * (lens_col list)
+and lens_col      = {
```

Comment:
Not sure I understand the reason for this type and using lens_col list instead of typ (where typ has to be a record / row type).  Could we use record/row types here, but disallow open rows in typechecking, like with tables?  Also not sure what "table" and "alias" are.

Reply:
Columns have the name `alias` in links, and map to the table called `table` in the database which has a column `name`. This allows taking two tables with the same columns, and giving them differing aliases for instance.